src/Particle/DistanceTableData.h

```diff
@@ -255,17 +270,67 @@ struct DistanceTableData
   }
 #endif
 
-  ///evaluate the full Distance Table
-  virtual void evaluate(ParticleSet& P) = 0;
+  /** return full table distances
+   */
+  const std::vector<DistRowType>& getDistances() const { return Distances; }
 
-  /// evaluate the Distance Table for a given electron
-  virtual void evaluate(ParticleSet& P, int jat) = 0;
+  /** return full table displacements
+   */
+  const std::vector<DisplRowType>& getDisplacements() const { return Displacements; }
 
-  ///evaluate the temporary pair relations
-  virtual void move(const ParticleSet& P, const PosType& rnew) = 0;
+  /** return a row of distances for a given target particle
+   */
+  const DistRowType& getDistRow(int iel) const { return Distances[iel]; }
 
-  ///update the distance table by the pair relations
-  virtual void update(IndexType jat) = 0;
+  /** return a row of displacements for a given target particle
+   */
+  const DisplRowType& getDisplRow(int iel) const { return Displacements[iel]; }
+
+  /** return old distances set up by move() for optimized distance table consumers
+   */
+  virtual const DistRowType& getOldDists() const
+  {
+    APP_ABORT("DistanceTableData::getOldDists is used incorrectly! Contact developers on github.");
+    return Temp_r; // dummy return to avoid compiler warning.
+  }
+
+  /** return old displacements set up by move() for optimized distance table consumers
+   */
+  virtual const DisplRowType& getOldDispls() const
+  {
+    APP_ABORT("DistanceTableData::getOldDispls is used incorrectly! Contact developers on github.");
+    return Temp_dr; // dummy return to avoid compiler warning.
+  }
+
+  /** return the temporary distances when a move is proposed
+   */
+  const DistRowType& getTemporalDists() const { return Temp_r; }
```

Comment:
getTempDists() is clearer to me. Temporal is not a common synonym of temporary.

Reply:
All changed.